scripts/meta_analysis.py

```diff
@@ -59,11 +76,18 @@ def n_meta( studies : List[Tuple['Study','VariantData']], is_het_test = False ):
     else:
         het_p=None
     #TODO se
-    return ( beta_meta, None, max(sys.float_info.min * sys.float_info.epsilon, 2 * scipy.stats.norm.sf( abs( sum( effs_size ) ) / math.sqrt(tot_size) )), het_p)
-
+    return ( beta_meta, None, max(sys.float_info.min * sys.float_info.epsilon, 2 * scipy.stats.norm.sf( abs( sum( effs_size ) ) / math.sqrt(tot_size) )), het_p) if len(effs_size)==len(studies) else (None, None, None, None)
 
-def inv_var_meta( studies : List[Tuple['Study','VariantData']], is_het_test = False):
 
+def inv_var_meta( studies : List[Tuple['Study','VariantData']], is_het_test = False ) -> Tuple:
+    '''
+        Computes inverse-variance weighted meta-analysis for variants in studies
+        input:
+            studies: studies and data in tuples
+            is_het_test: boolean, do heterogenity test
+        output:
+            tuple with results from meta-analysis and heterogeneity test p-value (if calculated)
+    '''
```

Comment:
can you actually move the het test out of this function.  Each function should do only one thing. You can add the het test e.g. into  do_meta

Reply:
Sure, sounds like a good improvement.